### PR TITLE
Remove user prompts from host setup

### PR DESCRIPTION
**Pull request by vdanix**
PR#367 ported to Android 12 SMR0/Branch

Tracked-On: OAM-105716
Signed-off-by: vdanix <vishwanathx.dani@intel.com>

### CODE REVIEW

**Review by basilchew — CHANGES_REQUESTED** [review]
The PR is wrong, as it is adding the patch file instead of patching the script.
Please check.

**Comment by vdanix** [comment]
> 

updated the pr

**Review by basilchew — APPROVED** [review]

File: bsp_diff/caas/device/intel/civ/host/vm-manager/0040-Remove-user-prompts-from-host-setup.patch [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+@@ -197,6 +193,12 @@ function install_vm_manager() {
+ }
+ 
+ function ubu_enable_host_gvt(){
++    if [[ ! `cat /etc/default/grub` =~ "i915.enable_gvt=1 intel_iommu=on i915.force_probe=*" ]]; then
++        sed -i "s/GRUB_CMDLINE_LINUX=\"/GRUB_CMDLINE_LINUX=\"i915.enable_gvt=1 intel_iommu=on i915.force_probe=*/g" /etc/default/grub
++        update-grub
++
++        echo -e "\nkvmgt\nvfio-iommu-type1\nvfio-mdev\n" >> /etc/initramfs-tools/modules
++        update-initramfs -u -k all
```
Comment (basilchew) [inline bsp_diff/caas/device/intel/civ/host/vm-manager/0040-Remove-user-prompts-from-host-setup.patch:38]: This section is wrong, please check.

**Review by basilchew — CHANGES_REQUESTED** [review]

File: bsp_diff/caas/device/intel/civ/host/vm-manager/0040-Remove-user-prompts-from-host-setup.patch (hunk 1 above)
Comment (vdanix) [reply bsp_diff/caas/device/intel/civ/host/vm-manager/0040-Remove-user-prompts-from-host-setup.patch:38]: @basilchew  we reslove conflict and these changes are from s/mr0/stable https://github.com/projectceladon/vm_manager/blob/celadon/s/mr0/stable/scripts/setup_host.sh#L260

Comment (basilchew) [reply bsp_diff/caas/device/intel/civ/host/vm-manager/0040-Remove-user-prompts-from-host-setup.patch:38]: If you apply the patch, you get this:

```
function ubu_enable_host_gvt(){
    if [[ ! `cat /etc/default/grub` =~ "i915.enable_gvt=1 intel_iommu=on i915.force_probe=*" ]]; then
        sed -i "s/GRUB_CMDLINE_LINUX=\"/GRUB_CMDLINE_LINUX=\"i915.enable_gvt=1 intel_iommu=on i915.force_probe=*/g" /etc/default/grub
        update-grub

        echo -e "\nkvmgt\nvfio-iommu-type1\nvfio-mdev\n" >> /etc/initramfs-tools/modules
        update-initramfs -u -k all

    if [[ ! `cat /etc/default/grub` =~ "i915.enable_guc="(0x)?0*"7" ]] &&
       [[ ! `cat /etc/default/grub` =~ "i915.enable_gvt=1" ]]; then

        read -p "The grub entry in '/etc/default/grub' will be updated for enabling GVT-g and GVT-d, do you want to continue? [Y/n]" res
        if [ x$res = xn ]; then
            return
        fi
        if [[ ! `cat /etc/default/grub` =~ "intel_iommu=on i915.force_probe=*" ]]; then
            sed -i "s/GRUB_CMDLINE_LINUX=\"/GRUB_CMDLINE_LINUX=\"intel_iommu=on i915.force_probe=* /g" /etc/default/grub
        fi
        sed -i "s/GRUB_CMDLINE_LINUX=\"/GRUB_CMDLINE_LINUX=\"i915.enable_gvt=1 /g" /etc/default/grub
        update-grub

        echo -e "\nkvmgt\nvfio-iommu-type1\nvfio-mdev\n" >> /etc/initramfs-tools/modules
        update-initramfs -u -k all

        reboot_required=1
    fi
}

```

1) The first "if" statement is open-ended, it is missing the closing "fi"
2) If you check the logic, the second u"if" statement already covers whatever the first "if" statement is trying to do.

Therefore, please remove the first "if" statement and related code.

**Review by basilchew — CHANGES_REQUESTED** [review]
This does not apply cleanly:

```
~/workspace/civ_s_mr0/device/intel/civ/host/vm-manager/scripts ((bc54bca...)) : git am -3 ~/workspace/civ_s_mr0/vendor/intel/utils_vertical/bsp_diff/caas/device/intel/civ/host/vm-manager/0040-Remove-user-prompts-from-host-setup.patch
Applying: Remove user prompts from host setup
Using index info to reconstruct a base tree...
M       scripts/setup_host.sh
Falling back to patching base and 3-way merge...
Auto-merging scripts/setup_host.sh
CONFLICT (content): Merge conflict in scripts/setup_host.sh
Recorded preimage for 'scripts/setup_host.sh'
error: Failed to merge in the changes.
Patch failed at 0001 Remove user prompts from host setup
Use 'git am --show-current-patch' to see the failed patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".
~/workspace/civ_s_mr0/device/intel/civ/host/vm-manager/scripts ((bc54bca...) *+|AM 1/1) : git diff
diff --cc scripts/setup_host.sh
index 7223f17,5912f9d..0000000
--- a/scripts/setup_host.sh
+++ b/scripts/setup_host.sh
@@@ -197,7 -193,6 +197,10 @@@ function install_vm_manager() 
  }
  
  function ubu_enable_host_gvt(){
++<<<<<<< HEAD
 +
++=======
++>>>>>>> Remove user prompts from host setup
      if [[ ! `cat /etc/default/grub` =~ "i915.enable_guc="(0x)?0*"7" ]] &&
         [[ ! `cat /etc/default/grub` =~ "i915.enable_gvt=1" ]]; then

```

Please check

**Pull request merged**